service/proxy/src/test/java/org/kaazing/gateway/service/proxy/ProxyServiceExtensionTest.java

```diff
@@ -0,0 +1,117 @@
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.kaazing.gateway.service.proxy;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.mina.core.session.IoSession;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.gateway.service.ServiceProperties;
+import org.kaazing.mina.core.session.IoSessionEx;
+
+public class ProxyServiceExtensionTest {
+
+    private static CountDownLatch latch;
+
+    @Rule
+    public GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("tcp://localhost:8888"))
+                            .connect(URI.create("tcp://localhost:8051"))
+                            .type("proxy")
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    @Before
+    public void init() {
+        latch = new CountDownLatch(3);
+    }
+
+    @Test
+    public void shouldInvokeExtension() throws Exception {
+        CountDownLatch backendReady = new CountDownLatch(1);
+
+        Thread t = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    ServerSocket backendServer = new ServerSocket();
+                    backendServer.bind(new InetSocketAddress("localhost", 8051));
+                    backendReady.countDown();
+                    Socket s = backendServer.accept();
+                    s.close();
+                    backendServer.close();
+                } catch (Exception ex) {
+                    fail("Unexpected exception in backend server: " + ex);
+                }
+            }
+        }, "backendServer-thread");
+        t.start();
+
+        boolean backendBound = backendReady.await(1, TimeUnit.SECONDS);
+        assertTrue("Backend Server failed to bind, pre-conditions not established", backendBound);
+
+        // connect to the service to ensure the extension is executed
+        Socket clientSocket = new Socket("localhost", 8888);
```

Comment:
See if you can use ARM block (try with resources syntax). Same goes with ServerSocket in the above